doc: add documentation describing CI internals (#394)
nim-works/nimskull: doc/docs.rst

- | `Style guide <style_guide.html>`_
  | Stanard library and compiler style guide

- | `Continuous integration internals <ci.html>`_

>   | The documentation describes how CI is implemented and common troubleshooting steps for failures. Recommended read if you are looking to work on the CI infrastructure.

Reviewer: ```suggestion
  | Describes how CI is implemented and steps for troubleshooting common failures. Required reading for those looking to work on the CI infrastructure.
```
